### PR TITLE
feat(geoprocessing): allow workers to use di

### PR DESCRIPTION
**Pull request by kgajowy**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

**Comment by kgajowy** [comment]
Thanks @Dyostiq for helping out with this one!

**Review by hotzevzl — APPROVED** [review]

File: geoprocessing/src/modules/worker/worker-resolver.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  ): Worker<Input, Output> {
+    if (this._worker) {
+      throw new Error('Worker is already created!');
+    }
+    this._worker = new Worker<Input, Output>(
+      queueName,
+      (job: Job) => processor.process(job),
+      this.config.redis,
+    );
+    return this._worker as Worker<Input, Output>;
```
Comment (Dyostiq) [inline geoprocessing/src/modules/worker/worker-resolver.ts:26]:
```suggestion
    const worker = new Worker<Input, Output>(
      queueName,
      (job: Job) => processor.process(job),
      this.config.redis,
    );
    this._worker = worker;
    return worker;
```
no cast needed :)

File: geoprocessing/test/worker-module/worker-module-di-usage.e2e-spec.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+let app: TestingModule;
+let queue: Queue;
+let processor: ExampleProcessingService;
+
+const queueName = 'test-queue';
+const initValue = 5;
+const addValue = 3;
+
+/**
+ * This tests USES redis to ensure that workers module works with full flow
```
Comment (Dyostiq) [inline geoprocessing/test/worker-module/worker-module-di-usage.e2e-spec.ts:20]: 🚀

File: geoprocessing/src/modules/worker/worker-resolver.ts (hunk 1 above)
Comment (Dyostiq) [inline geoprocessing/src/modules/worker/worker-resolver.ts:26]: If the instance is disposable, and I am not sure if it's a wrapper, or it is a resolver (the naming is mixed in the PR a bit), I'd call it a builder, as builders have a single-use nature and shouldn't be reused after `build`.

**Review by Dyostiq — APPROVED** [review]

File: geoprocessing/src/modules/worker/worker-resolver.ts (hunk 1 above)
Comment (kgajowy) [reply geoprocessing/src/modules/worker/worker-resolver.ts]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/202/commits/d4a3c87a04836695481fb7e64cc6eaa76d5bdcb2

Comment (kgajowy) [reply geoprocessing/src/modules/worker/worker-resolver.ts]: fixed in https://github.com/Vizzuality/marxan-cloud/pull/202/commits/d4a3c87a04836695481fb7e64cc6eaa76d5bdcb2

**Pull request merged**